### PR TITLE
iRPC_2D

### PR DESCRIPTION
**Pull request by bapavlov**
Small changes to the digitizer to enable 2D readout for iRPC. The option is configurable with a flag in the configuration file.

### CODE REVIEW

File: SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc [hunk 1]
```diff
@@ -284,3 +313,36 @@ int RPCSimModelTiming::getClSize(uint32_t id,float posX, CLHEP::HepRandomEngine*
...
+  //  std::cout<<"((hit_pos.x())\t"<<(hit_pos.x())<<std::endl;
+
+//  double deltaw = fabs((roll.centreOfStrip(strip)).x()-hit_pos.x());
+  double deltawL = fabs((roll.centreOfStrip(leftStrip)).x()-hit_pos.x());
+  double deltawR = fabs((roll.centreOfStrip(rightStrip)).x()-hit_pos.x());
+
+  //  std::cout<<"deltaw\t"<<deltaw<<std::endl;
+  //  std::cout<<"deltawL\t"<<deltawL<<std::endl;
+  //  std::cout<<"deltawR\t"<<deltawR<<std::endl;
+
```
Comment (kpedro88) [inline SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc]: please remove commented lines of code if not needed

File: SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc [hunk 2]
```diff
@@ -284,3 +313,32 @@ int RPCSimModelTiming::getClSize(uint32_t id,float posX, CLHEP::HepRandomEngine*
...
+  int leftStrip =  strip-1;
+  int rightStrip = strip+1;
+
+  if(leftStrip<0)
+    return +1;
+  if( rightStrip > roll.nstrips())
+    return -1;
+
+  //  std::cout<<"((roll.centreOfStrip(strip)).x())\t"<<((roll.centreOfStrip(strip)).x())<<std::endl;
+  //  std::cout<<"((hit_pos.x())\t"<<(hit_pos.x())<<std::endl;
```
Comment (kpedro88) [inline SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc]: please remove

File: SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc [hunk 3]
```diff
@@ -284,3 +313,32 @@ int RPCSimModelTiming::getClSize(uint32_t id,float posX, CLHEP::HepRandomEngine*
...
+
+//  double deltaw = fabs((roll.centreOfStrip(strip)).x()-hit_pos.x());
```
Comment (kpedro88) [inline SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc]: please remove

File: SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc [hunk 4]
```diff
@@ -226,6 +244,17 @@ RPCDetId rpcId = roll->id();
...
             RPCDigi adigi(j+1,time_hit);
+	    adigi.hasTime(true);
+            adigi.setTime(precise_time);
+   if(do_Y)
+	{
+     double positionY = CLHEP::RandFlat::shoot(engine,striplength);
+positionY-=striplength/2; 
+	  adigi.hasY(true);
+	  adigi.setY(positionY);
+//          adigi.setDeltaY(striplength/sqrt(12.));
```
Comment (kpedro88) [inline SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc]: please remove

File: SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc [hunk 5]
```diff
@@ -133,8 +138,9 @@ void RPCSimModelTiming::simulate(const RPCRoll* roll,
 	if (clsize%2 == 0 ){
 	  // insert the last strip according to the 
 	  // simhit position in the central strip 
-	  double deltaw=roll->centreOfStrip(centralStrip).x()-entr.x();
-	  if (deltaw<0.) {
+	  int lr = LeftRightNeighbour(*roll, entr, centralStrip);
+	  //	  std::cout<<"LeftRightNeighbour(*roll, entr, centralStrip)\t"<<lr<<std::endl;
```
Comment (kpedro88) [inline SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc]: please remove

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
@smuzaffar can you check if there is a problematic node?

**Comment by davidlange6** [comment]
please test

likely cvmfs issue

**Comment by smuzaffar** [comment]
please test
@kpedro88 , cvmfs cache was corrupted on vocms0302. This should be fixed now.

**Comment by civanch** [comment]
+1

File: SimMuon/RPCDigitizer/python/muonRPCDigis_cfi.py [hunk 6]
```diff
@@ -77,7 +77,9 @@
             Nbxing = cms.int32(9),
             timeJitter = cms.double(1.0),
             IRPC_time_resolution = cms.double(0.1),
-            IRPC_electronics_jitter = cms.double(0.025)
+            IRPC_electronics_jitter = cms.double(0.025),
+           sigmaY = cms.double(1.), #math.sqrt((0.025)*(0.025)+(0.025)*(0.025))*0.66*299792458*1e+2*1e-9 = 0.7
```
Comment (davidlange6) [inline SimMuon/RPCDigitizer/python/muonRPCDigis_cfi.py]: @bapavlov - please fix the indentation here.

File: SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc [hunk 7]
```diff
@@ -166,8 +173,14 @@ void RPCSimModelTiming::simulate(const RPCRoll* roll,
 	  RPCDigi adigi(*i,time_hit);
           adigi.hasTime(true);
           adigi.setTime(precise_time);
+          if(do_Y)
+          {
+		adigi.hasY(true);
+                adigi.setY(smearedPositionY);
```
Comment (davidlange6) [inline SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc]: and here

File: SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc [hunk 8]
```diff
@@ -177,35 +190,28 @@ void RPCSimModelTiming::simulate(const RPCRoll* roll,
...
-//std::cout<<"RPCSimModelTiming::simulateNoise X2"<<std::endl;
-// std::cout<<"rpcId.rawId() = "<<rpcId.rawId()<<std::endl;
   std::vector<float> vnoise = (getRPCSimSetUp())->getNoise(rpcId.rawId());
-//std::cout<<"RPCSimModelTiming::simulateNoise X3"<<std::endl;
   std::vector<float> veff = (getRPCSimSetUp())->getEff(rpcId.rawId());
-//std::cout<<"RPCSimModelTiming::simulateNoise X4"<<std::endl;
   unsigned int nstrips = roll->nstrips();
   double area = 0.0;
-
+float striplength, xmin,xmax ;
```
Comment (davidlange6) [inline SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc]: and here

File: SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc [hunk 9]
```diff
@@ -223,9 +229,17 @@ RPCDetId rpcId = roll->id();
       
       double precise_time = CLHEP::RandFlat::shoot(engine, (nbxing*gate)/gate);
       int time_hit = (static_cast<int>(precise_time)) - nbxing/2;
-//      std::pair<int, int> digi(j+1,time_hit);
-//      strips.insert(digi);
             RPCDigi adigi(j+1,time_hit);
+	    adigi.hasTime(true);
+            adigi.setTime(precise_time);
```
Comment (davidlange6) [inline SimMuon/RPCDigitizer/src/RPCSimModelTiming.cc]: and here

**Comment by civanch** [comment]
please test

**Comment by kpedro88** [comment]
please abort

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**